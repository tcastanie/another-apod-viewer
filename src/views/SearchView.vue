<template>
  SEARCH
  <SearchInput @search-results="updateList" />
  <SearchList :list="apodList" />
</template>

<script>
import SearchInput from "../components/SearchInput.vue";
import SearchList from "../components/SearchList.vue";
export default {
  components: {
    SearchInput,
    SearchList,
  },
  data() {
    return {
      apodList: [],
    };
  },
  computed: {
    dateToday() {
      const date = new Date();
      return date.toISOString().split("T")[0];
    },
  },
  methods: {
    updateList(e) {
      this.apodList = e;
    },
  },
};
</script>

<style lang="scss" scoped>
.v-search {
  position: relative;
}
</style>
