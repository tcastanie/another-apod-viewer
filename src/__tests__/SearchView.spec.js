import { describe, it, expect, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import { routes } from "@/router/index.js";
import SearchView from "@/views/SearchView.vue";
import { testList } from "./constantsMockups";

let router;
beforeEach(async () => {
  router = createRouter({
    history: createWebHistory(),
    routes: routes,
  });
});

describe("SearchView", async () => {
  it("renders properly", async () => {
    router.push("/search");
    await router.isReady();
    const wrapper = mount(SearchView, {
      global: {
        plugins: [router],
        stubs: {
          "search-input": true,
          "search-list": true,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders properly", async () => {
    router.push("/search");
    await router.isReady();
    const wrapper = mount(SearchView, {
      global: {
        plugins: [router],
        stubs: {
          "search-input": true,
          "search-list": true,
        },
      },
    });
    wrapper.vm.updateList(testList);
    expect(wrapper.vm.apodList).toEqual(testList);
  });
});
