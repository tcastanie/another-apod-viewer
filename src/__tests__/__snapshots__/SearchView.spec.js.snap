// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchView > renders properly 1`] = `
"<h1>Search</h1>
<search-input-stub></search-input-stub>
<search-list-stub list=\\"\\"></search-list-stub>"
`;
